app/controllers/api/v1/appointment_slots_controller.rb

```diff
@@ -0,0 +1,87 @@
+module Api
+  module V1
+    class AppointmentSlotsController < Api::V1::ApiController
+      load_and_authorize_resource :appointment_slot, parent: false
+
+      resource_description do
+        short 'Manage a list of appointment quotas for special days'
+        formats ['json']
+        error 401, "Unauthorized"
+        error 404, "Not Found"
+        error 422, "Validation Error"
+        error 500, "Internal Server Error"
+      end
+
+      def_param_group :appointment_slot do
+        param :appointment_slot, Hash, required: true do
+          param :timestamp, String
+          param :note, String
+          param :quota, :number
+        end
+      end
+
+      def appointment_slot_params
+        params.require(:appointment_slot).permit([:timestamp, :quota, :note])
+      end
+
+      api :GET, '/v1/appointment_slots', "List upcoming appointment slots"
+      def index
+        render_with_timezone(@appointment_slots.upcoming.ascending)
+      end
+
+      api :GET, '/v1/appointment_slots/calendar', "List upcoming appointment slots aggregated by dates"
+      def calendar
+        from = params[:from] ? Date.parse(params[:from]) : Date.today
+        to = Date.parse(params[:to])
+        render json: AppointmentSlot.calendar(from, to).to_json, status: 200
+      end
+
+      api :POST, "/v1/appointment_slots", "Add an appointment slot"
+      param_group :appointment_slot
+      def create
+        @appointment_slot.timestamp = @appointment_slot.timestamp.change(sec: 0)
```

Comment:
nice! I didn't know about ```change```

Reply:
Yup. Had issues in comparing timestamps. So I'm making sure seconds aren't saved